src/compas_fab/robots/trajectory.py

```diff
@@ -149,6 +149,99 @@ def data(self, data):
         self.effort = data.get('effort') or []
         self.time_from_start = Duration.from_data(data.get('time_from_start') or {})
 
+    def copy(self):
+        """Create a copy of this :class:`JointTrajectoryPoint`.
+
+        Returns
+        -------
+        :class:`JointTrajectoryPoint`
+            An instance of :class:`JointTrajectoryPoint`
+        """
+        cls = type(self)
+        return cls.from_data(self.data)
+
+    @property
+    def velocity_states(self):
+        """A dictionary of joint velocities by joint name."""
+        self.check_joint_names()
+        return dict(zip(self.joint_names, self.velocities))
+
+    @property
+    def acceleration_states(self):
+        """A dictionary of joint accelerations by joint name."""
+        self.check_joint_names()
+        return dict(zip(self.joint_names, self.accelerations))
+
+    @property
+    def effort_states(self):
+        """A dictionary of joint efforts by joint name."""
+        self.check_joint_names()
+        return dict(zip(self.joint_names, self.effort))
+
+    def merge(self, other):
+        """Merge the `JointTrajectoryPoint` with another `JointTrajectoryPoint` in place
+        along joint names.  The other `JointTrajectoryPoint` takes precedence over this
+        `JointTrajectoryPoint` in case a joint value is present in both.
+        Caution: `joint_names` may be rearranged.
```

Comment:
```suggestion
        """Merge the ``JointTrajectoryPoint`` with another ``JointTrajectoryPoint`` in place
        along joint names.  The other ``JointTrajectoryPoint`` takes precedence over this
        ``JointTrajectoryPoint`` in case a joint value is present in both.

        Note
        ----
                Caution: ``joint_names`` may be rearranged.
```